upt_macports/upt_macports.py

```diff
@@ -278,3 +278,32 @@ def package_versions(self, name):
             sys.exit(f'The command "{cmd}" failed. '
                      'Please make sure you have MacPorts installed '
                      'and/or your PATH is set-up correctly.')
+
+    def cpan_version(self, vers):
+        if vers[0] == "v":
+            start = 1
+        else:
+            start = 0
+
+        index = vers.find('.')
+        other_dot = vers[index+1:].find('.')
+        if index == -1 or other_dot != -1:
+            return vers[start:]
+        ret = vers[start:index]
+        index += 1
+        fractional = vers[index:]
+        index = 0
+        while index < len(fractional) or index < 6:
+            sub = fractional[index:index+3]
+            if len(sub) < 3:
+                sub += '0'*(3-len(sub))
+            ret = ret + '.' + str(int(sub))
+            index += 3
+        return ret
+
+    def needs_requirement(self, req, phase, frontend):
+        if frontend == 'cpan' and req.specifier:
+            print(req.specifier)
```

Comment:
in the "final" version we don't really need a print-statement anymore - correct?

Reply:
Ohh sorry.